pkg/streamingpromql/functions/common.go

```diff
@@ -26,11 +26,11 @@ var DropSeriesName = SeriesMetadataFunctionDefinition{
 }
 
 // InstantVectorSeriesFunction is a function that takes in an instant vector and produces an instant vector.
-type InstantVectorSeriesFunction func(seriesData types.InstantVectorSeriesData, memoryConsumptionTracker *limiting.MemoryConsumptionTracker) (types.InstantVectorSeriesData, error)
+type InstantVectorSeriesFunction func(seriesData types.InstantVectorSeriesData, scalarArgsData []types.ScalarData, memoryConsumptionTracker *limiting.MemoryConsumptionTracker) (types.InstantVectorSeriesData, error)
```

Comment:
[nit] Maybe `scalarArgs` is enough? Not sure the `Data` is necessary.

Reply:
I wanted to differentiate between the processed args, and the `ScalarOperator` args in `FunctionOverInstantVector`.